src/dotty/tools/dotc/transform/FullParameterization.scala

```diff
@@ -176,7 +176,12 @@ trait FullParameterization {
           } else EmptyTree
         }
         tree match {
-          case Ident(_) => rewireCall(thisRef)
+          case Ident(_) =>
+            val rewired = rewiredTarget(tree, derived)
+            if (rewired.exists) // This only happens when `tree` is the `from` part of a Return
```

Comment:
If `Ident`s inside `Return`s need special handling, I'd prefer to handle them on `Return` node level. IE add `case t: Return => ` which has special handling of return.